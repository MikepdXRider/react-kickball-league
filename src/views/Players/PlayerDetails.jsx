import React from 'react'
import { useParams } from 'react-router'
import { useEffect, useState } from 'react/cjs/react.development';
import PlayerCard from '../../components/PlayerCard/PlayerCard.jsx';
import { getPlayerById } from '../../services/players.js';

export default function PlayerDetails() {
    const { id } = useParams();

    const [isLoading, setIsLoading] = useState(true);
    const [playerDataObj, setPlayerDataObj] = useState({});


    useEffect(()=> {
        async function getPlayerData() {
            const newPlayerDataObj = await getPlayerById(id);
            setPlayerDataObj(newPlayerDataObj);
            setIsLoading(false);
        }

        getPlayerData();
    }, [id])


    return (
        <main>
            {
                isLoading
                ? <h1>Loading...</h1>
                : <PlayerCard playerDataObj={playerDataObj} />
            }
        </main>
    )
}
