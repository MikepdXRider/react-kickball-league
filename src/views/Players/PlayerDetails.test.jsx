import { screen } from "@testing-library/react";
import { render } from "@testing-library/react";
import { MemoryRouter, Route } from "react-router";
import PlayerDetails from "./PlayerDetails.jsx";


it ('renders PlayerDetils', async () => {
    const { container } = render(
    <MemoryRouter initialEntries={[`/players/1`]}>  
        <Route path='/players/:id'>
            <PlayerDetails />
        </Route>
    </MemoryRouter>
    )
    const title = await screen.findByText(/Ben E. Jetts/i);

    expect(title).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})