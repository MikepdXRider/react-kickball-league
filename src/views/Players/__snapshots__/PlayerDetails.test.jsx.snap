// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PlayerDetils 1`] = `
<div>
  <main>
    <article>
      <h3>
        Ben E. Jetts
      </h3>
      <h6>
        2021-11-22T20:11:15+00:00
      </h6>
      <h5>
        Position: 
        Pitcher
      </h5>
    </article>
  </main>
</div>
`;
